### PR TITLE
[G2M] change Pyramid datatransform data access

### PR DESCRIPTION
**Pull request by kc-leung**
- Changed Pyramid chart data manipulation to match other charts logic
- Added numeric formatting for bar & pyramid chart (ex: 10000 -> 10k, 1000000 -> 1M) using d3 formatting

### CODE REVIEW

**Review by woozyking — COMMENTED** [review]
where's the logic that appends the k/m suffixes for larger numbers?

**Comment by kc-leung** [comment]
> where's the logic that appends the k/m suffixes for larger numbers?

I am just using d3.format function to do the suffixes. Is there any plan to remove d3 in the future from chart-system? If so I will add a function to do the formatting

**Comment by woozyking** [comment]
> > where's the logic that appends the k/m suffixes for larger numbers?
> 
> 
> 
> I am just using d3.format function to do the suffixes. Is there any plan to remove d3 in the future from chart-system? If so I will add a function to do the formatting 

Oh I see. No, d3's good. But keep the option open for the upcoming localization development (d3 should be able to handle that if needed)

**Review by woozyking — APPROVED** [review]

File: src/util/numeric.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @returns { boolean } - return true if it is only intiger numbers in the array, false otherwise
+ */
+export const onlyNumbers = (array) => {
+  return array.every(el => {
+    return typeof el === 'number'
+  })
+}
+
+/**
+ * getRoundToNumberDigit - rounds a number up to the next largest integer based of the lenght of its digits
```
Comment (geoerika) [inline src/util/numeric.js:13]: "length" - I am doing this all the time myself.

**Review by geoerika — APPROVED** [review]
Just a few typos...

**Pull request merged**